packages/models-library/src/models_library/user_preferences.py

```diff
@@ -0,0 +1,186 @@
+from collections.abc import Mapping
+from enum import Enum, auto
+from typing import Any, ClassVar, TypeAlias
+
+from pydantic import BaseModel, Field
+from pydantic.main import ModelMetaclass
+
+from .services import ServiceKey
+from .services_ui import WidgetType
+from .utils.enums import StrAutoEnum
+
+IntStr: TypeAlias = int | str
+DictStrAny = dict[str, Any]
+AbstractSetIntStr: TypeAlias = set[IntStr]
+MappingIntStrAny: TypeAlias = Mapping[IntStr, Any]
+
+
+class _AutoRegisterMeta(ModelMetaclass):
+    _registered_user_preference_classes: ClassVar[dict[str, type]] = {}
+
+    def __new__(cls, name, bases, attrs):
+        new_class = super().__new__(cls, name, bases, attrs)
+
+        if name != BaseModel.__name__:
+            if name in cls._registered_user_preference_classes:
+                msg = (
+                    f"Class named '{name}' was already defined at "
+                    f"{cls._registered_user_preference_classes[name]}."
+                    " Please choose a different class name!"
+                )
+                raise TypeError(msg)
+            cls._registered_user_preference_classes[name] = new_class
+
+        return new_class
+
+
+PreferenceName: TypeAlias = str
+
+
+class _ExtendedBaseModel(BaseModel, metaclass=_AutoRegisterMeta):
+    ...
+
+
+class ValueType(str, Enum):
```

Comment:
Since you do not use `auto`, I interpret that this enum is used to map pre-defined values ("boolean", etc) to your own keys.
TIP: I like to express this using `Listeral["boolean", "string", ... ]`.

Reply:
If I got this you suggest dropping the enum? I would like to keep it as it is pretty convenient and I don't have to redefine all the values in different places

Reply:
no longer used